File: dsa/sorting/insertion.py
a=[5,3,6,9,23,1]
for i in range(len(a)):
    temp=a[i]
    j=i-1
    while(j>=0 and a[j]>temp):
        a[j+1]=a[j]
        j-=1
    a[j+1]=temp
print(a)

File: dsa/tree/sorting/Insertion.java
public class Insertion{
    public static void main(String[] args) {
        int[] arr={1,23,4,7,3};
        for (int i = 0; i < arr.length; i++) {
            int temp=arr[i];
            int j=i-1;
            while(j>=0 && arr[j]>temp){
                arr[j+1]=arr[j];
                j--;
            }
            arr[j+1]=temp;
        }
        for(int i:arr){
            System.out.println(i);
        }
    }
}